private/bufpkg/bufconnect/interceptors.go

```diff
@@ -28,17 +30,33 @@ const (
 // NewSetCLIVersionInterceptor returns a new Connect Interceptor that sets the Buf CLI version into all request headers
 func NewSetCLIVersionInterceptor(version string) connect.UnaryInterceptorFunc {
 	interceptor := func(next connect.UnaryFunc) connect.UnaryFunc {
-		return func(
-			ctx context.Context,
-			req connect.AnyRequest,
-		) (connect.AnyResponse, error) {
+		return func(ctx context.Context, req connect.AnyRequest) (connect.AnyResponse, error) {
 			req.Header().Set(CliVersionHeaderName, version)
 			return next(ctx, req)
 		}
 	}
 	return interceptor
 }
 
+// NewCLIWarningInterceptor returns a new Connect Interceptor that logs CLI warnings returned by server responses.
+func NewCLIWarningInterceptor(container applog.Container) connect.UnaryInterceptorFunc {
+	interceptor := func(next connect.UnaryFunc) connect.UnaryFunc {
+		return func(ctx context.Context, req connect.AnyRequest) (connect.AnyResponse, error) {
+			resp, err := next(ctx, req)
+			if resp != nil && resp.Header() != nil {
+				encoded := resp.Header().Get(CLIWarningHeaderName)
```

Comment:
Would `.Values()` be more appropriate? If our goal is to be future-compatible, perhaps we should expect multiple headers?